uninstall.php

```diff
@@ -8,39 +8,133 @@
  * @package Airstory
  */
 
-namespace Airstory;
+namespace Airstory\Uninstall;
+
+use Airstory\Connection as Connection;
 
 require_once __DIR__ . '/includes/connection.php';
 
-global $wpdb;
+/**
+ * Retrieve the IDs of any sites with active Airstory connections.
+ *
+ * @global $wpdb
+ *
+ * @return array An array of WordPress site IDs that have one or more active Airstory connections.
+ */
+function get_active_site_ids() {
+	global $wpdb;
+
+	/*
+	 * The SQL query attempts to parse a site ID out of the key name within wp_usermeta.
+	 *
+	 * Assuming the default WordPress $table_prefix (as set in wp-config.php) of "wp_", keys will
+	 * look like:
+	 *
+	 * - wp__airstory_target
+	 * - wp_2__airstory_target
+	 * - wp_3__airstory_target
+	 *
+	 * The SUBSTRING() function within MySQL accepts three arguments: the string, the starting point
+	 * (starting at 0), and the length of the string to return.
+	 *
+	 * The meta_key will be our string, and the length of $table_prefix (available via the
+	 * $wpdb->base_prefix property) + 1 gives us a starting point (e.g. everything before the first
+	 * digit of the site ID).
+	 *
+	 * The third argument is where things get more complicated: we need the length of the meta_key,
+	 * less the length of the static portion ("_airstory_target"), the length of the table prefix,
+	 * and one more (for the extra underscore immediately following the site ID).
+	 */
+	$site_ids = $wpdb->get_col( $wpdb->prepare( "
```

Comment:
we aren't storing a single user meta key of an array of site ID's that are connected to said user?

can't we store an array of IDs whenever they're added / removed / modified, making it a meta key check instead of a SQL query?